# TS definitions are to be interpreted by the agent node. The LTL core planner only uses the node names, connections, weights, and action names to build its automaton.
#
# Possible TS types:
#   6d_jointspace_region (6d joint position: joint name, joint position)
#   hebi_load

# Need test. Circular type could be changed. related to region_6d_jointspace_monitor.py under tl_automaton_std_transition_systems package

state_dim: ["6d_jointspace_region", "hebi_load"]
state_models:
    6d_jointspace_region:
        ts_type: "6d_jointspace_region"
        initial: "home"
        nodes: 
            home:
                attr:
                    type: "circular"
                    position: [0.4, 1.25, 1.61, -1.10, 1.62, 0.34]
                    radius: 0.8
                    hysteresis: 0.05
                connected_to:
                    home: "goto_home"
                    pick_ready: "goto_pick_ready" 
                    drop_ready: "goto_drop_ready"
            pick_ready:
                attr:
                    type: "circular"
                    position: [-0.45, 1.25, 1.61, -1.10, 1.62, 0.34]
                    radius: 0.8
                    hysteresis: 0.05
                connected_to:
                    pick_ready: "goto_pick_ready"
                    home: "goto_home"
            drop_ready:
                attr:
                    type: "circular"
                    position: [1.2, 1.25, 1.61, -1.10, 1.62, 0.34]
                    radius: 0.8
                    hysteresis: 0.05
                connected_to:
                    home: "goto_home" 
                    drop_ready: "goto_drop_ready"
    hebi_load:
        ts_type: "hebi_load"
        initial: "loaded"
        nodes:
            unloaded:
                connected_to:
                    loaded: "pick"
                    # unloaded: "idle"
            loaded:
                connected_to:
                    # loaded: "idle"
                    unloaded: "drop"

actions:
    goto_home:
        type: "move"
        weight: 10
        guard: "1"
        attr:
            jointposition: [0.4, 1.25, 1.61, -1.10, 1.62, 0.34]
    goto_pick_ready:
        type: "move"
        weight: 10
        guard: "1"
        attr:
            jointposition: [-0.45, 1.25, 1.61, -1.10, 1.62, 0.34]
    goto_drop_ready:
        type: "move"
        weight: 10
        guard: "1"
        attr:
            jointposition: [1.2, 1.25, 1.61, -1.10, 1.62, 0.34]
    pick:
        type: "hebi_pick"
        weight: 10
        guard: "pick_ready"
        attr:
            jointpositionlist: [[-0.45, 1.25, 1.61, -1.10, 1.62, 0.34],[-0.45, 0.88, 2.00, -0.34, 1.62, 0.34]]
    drop:
        type: "hebi_drop"
        weight: 10
        guard: "drop_ready"
        attr:
            jointpositionlist: [[1.2, 1.25, 1.61, -1.10, 1.62, 0.34],[1.2, 0.88, 2.00, -0.34, 1.62, 0.34]]
    # idle:
    #     type: "hebi_idle"
    #     weight: 10
    #     guard: "1"
             

